aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -554,13 +491,13 @@ fn native_read_derived_string(
     debug_assert_eq!(args.len(), 1);
     context.charge(AGGREGATOR_V2_READ_SNAPSHOT_BASE)?;
 
-    let derived_string_value = get_derived_string_field(&safely_pop_arg!(args, StructRef))?;
-
     let result_value = if let Some((resolver, mut delayed_field_data)) = get_context_data(context) {
-        let delayed_id = derived_string_value_field_as_id(derived_string_value, resolver)?;
-        delayed_field_data.read_derived(delayed_id, resolver)?
+        let derived_string_snapshot = safely_pop_arg!(args, Reference);
```

Comment:
Why is it a reference in this case / how does it differ from StructRef?. Why can't the StructRef be passed the same way just interpreted as id in the end (to value then to id)?

Reply:
Move values special case structs, so we have `StructRef` and `Reference` (generic). Here we don't have a struct anymore, only a `ValueImpl::Delayed`, and so the argument passed is a reference to it, i.e. `Reference`. It can work other way around though, if we use `Reference` instead of `StructRef`.